src/plasma/plasma_manager.cc

```diff
@@ -158,7 +161,8 @@ typedef struct {
   /** The object requests for this wait request. Each object request has a
    *  status field which is either PLASMA_QUERY_LOCAL or PLASMA_QUERY_ANYWHERE.
    */
-  ObjectRequest *object_requests;
+  std::unordered_map<ObjectID, ObjectRequest, decltype(&hashObjectID)>
+      *object_requests;
```

Comment:
Let's make this a "std::unordered_map" instead of a "pointer to a std::unordered_map" (and replace the malloc for WaitRequest by a new); this should work because we are only ever storing WaitRequest* in uthash data structures (as opposed to a full WaitRequest).